File: nodejs/20230429/hanoi.js
let readline = require("readline");
let rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

function hanoi_t(N, from, tmp, to) {
  if (N >= 1) {
    hanoi_t(N - 1, from, to, tmp);
    console.log(from + " " + to);
    hanoi_t(N - 1, tmp, from, to);
  }
  return;
}
function cnt(N) {
  if (N == 1) return 1;
  else return 2 ** N - 1;
}

let N;
let h_cnt = 0;
rl.on("line", function (line) {
  N = Number(line);
  h_cnt = cnt(N);
  console.log(h_cnt);
  hanoi_t(N, "1", "2", "3");
  rl.close();
});

File: python/20230507/hanoi.py

count = 0
def hanoi(n, a, b, c):
    global count
    if n>=1:
        count += 1
        hanoi(n-1, a, c, b)
        print(a, c)
        hanoi(n-1, b, a, c)

number = int(input())
print((2**number-1))
hanoi(number, '1', '2', '3')
print(count)
